pkg/nfd-gc/nfd-gc.go

```diff
@@ -172,7 +180,26 @@ func (n *nfdGarbageCollector) garbageCollect() {
 		klog.ErrorS(err, "failed to list NodeResourceTopology objects")
 	} else {
 		for _, nrt := range nrts.Items {
+			delete := false
+
 			if !nodeNames.Has(nrt.Name) {
+				delete = true
+			}
+
+			if label, ok := nrt.Labels[NRTOwnerPodLabel]; ok {
+				if s := strings.Split(string(label), "/"); len(s) == 2 {
```

Comment:
nit: no need for type conversion

```suggestion
				if s := strings.Split(label, "/"); len(s) == 2 {
```